### DISCUSSION TITLE
Named Groups

### DISCUSSION THREAD

**Discussion by jhealey5** [tailwindlabs/tailwindcss#7639] [Ideas]
Hello, I like to use group-hover but the lack of nesting makes me reach for js often. I know there's a tw plugin for nested groups but it's a bit much sometimes.

I'm wondering if it's possible to have named groups, much like the arbitrary variables, eg: `group-[mygroup]`. Then we could later use `group-[mygroup]-hover:opacity-100`?

Not sure if there's technical reasons it's not possible but it seems like an elegant solution if it could work.

**Comment by adamwathan** [comment]
Yeah I think this is a great idea and something we want to explore, I mentioned the same thing here:

https://github.com/tailwindlabs/tailwindcss/discussions/7547#discussioncomment-2217553

**Comment by jhealey5** [comment]
Ah brilliant, sorry I looked but must of missed that one. Great minds hey!

Looking forward to it! Cheers.

**Comment by onmax** [comment]
I just developed a new plugin very similar to this one that has all Tailwind 3 features.

Check it out here: https://github.com/onmax/tailwindcss-labeled-groups

**Comment by reinink** [comment]
Hey! Good news — we just added nested group and multiple peer support using variant modifiers in Tailwind CSS v3.2 🥳

You can read about it in our [announcement blog post](https://tailwindcss.com/blog/tailwindcss-v3-2#nested-group-and-multiple-peer-support-using-variant-modifiers), but for quick reference here’s what it looks like:

```html
<div class="group/sidebar ...">
  <!-- ... -->
  <div class="group/navitem ...">
    <a
      href="#"
      class="opacity-50 group-hover/sidebar:opacity-75 group-hover/navitem:bg-black/75"
    >
      <!-- ... -->
    </a>
  </div>
  <!-- ... -->
</div>
```

This lets you give each group a clear name that makes sense for that context on the fly, and Tailwind will generate the necessary CSS to make it work.

**Reply by pbiggar** [reply]
This is great. We just switched over to it (https://github.com/darklang/dark/pull/4545) and it works really nicely (esp without having to declare group names in advance). I previously spent a lot of time trying the various group plugins and this works better than both that I tried. Thanks so much!

**Reply by andreasvirkus** [reply]
Am I missing something? If you edit `group/rating` to be just `group` (same for the `group-hover/rating` rules), it works 🤔 

https://play.tailwindcss.com/hMkvNJyQwB

**Reply by wongjn** [reply]
You are missing something. Group names where [introduced in `v3.2`](https://tailwindcss.com/blog/tailwindcss-v3-2#nested-group-and-multiple-peer-support-using-variant-modifiers) – the Tailwind version in your Tailwind Play is set to `v2.2.19`.

**Reply by andreasvirkus** [reply]
Oh shoot, haha 😅 thanks for that... But it's also happening in my project with `3.4.1` 🤔

**Reply by wongjn** [reply]
Consider checking:
- Tailwind has compiled since you added the class.
- The `content` file globs sufficiently cover the file that the code appears in.
- [String concatenation/interpolation is not used to build the class name](https://tailwindcss.com/docs/content-configuration#dynamic-class-names). 
- The class name would resolve to a Tailwind class that Tailwind would generate a rule for.

Otherwise consider providing a project or [Tailwind Play](https://play.tailwindcss.com) that reproduces the unexpected behavior.

**Reply by sohel-silkhaus** [reply]
@reinink  I am not sure if this is right but when I don't have modifiers with named group, I need to use `&`. For example when I have `group` on the parent element, I can use something like `group:py-2` on its child. But when I have something like `group/item` on the parent element, I cannot use `group/item:py-2` on the child. Instead, I have to use `group-[&]/item:py-2` and then it works as intended. This is only when I don't have modifiers. Is this intended behaviour? It seems like bug to me. Or am I misunderstanding something?

**Reply by wongjn** [reply]
It is intended behavior. `group-*` variants are not really intended to be used this way. This can be evidenced by the fact that [there will be a new `in-*` variant in v4](https://github.com/tailwindlabs/tailwindcss/pull/15025) to address this usage.

**Comment by Jean-EmmanuelP** [comment]
Thank you so good to use it !

**Comment by hexxt-git** [comment]
this needs to be better highlighted in the documentation its really useful

**Reply by rozsazoltan** [reply]
Here: https://tailwindcss.com/docs/hover-focus-and-other-states#differentiating-nested-groups
